### PR TITLE
fix!: Fix for hierarchical firewall policy

### PR DESCRIPTION
**Pull request by renato-rudnicki**
The API resources used by `3-networks-dual-svpc/modules/hierarchical_firewall_policy` and `3-networks-hub-and-spoke/modules/hierarchical_firewall_policy` are deprecated and causing deployment failures on Foundation. This pull request addresses the issue by updating the resources to their latest versions.

Issue related: <https://github.com/hashicorp/terraform-provider-google/issues/20657>

### CODE REVIEW

File: 3-networks-hub-and-spoke/modules/hierarchical_firewall_policy/main.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2022 Google LLC
+ * Copyright 2021 Google LLC
```
Comment (apeabody) [inline 3-networks-hub-and-spoke/modules/hierarchical_firewall_policy/main.tf:2]: Is there a reason for this change?

**Comment by apeabody** [comment]
Thanks @renato-rudnicki - I suspect we also need to update the minimum version constraints:
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-networks-dual-svpc/modules/hierarchical_firewall_policy/versions.tf
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-networks-hub-and-spoke/modules/hierarchical_firewall_policy/versions.tf

For example `google_compute_firewall_policy`, wasn't added till [v3.82](https://github.com/hashicorp/terraform-provider-google/releases/tag/v3.82.0)

File: 3-networks-hub-and-spoke/modules/hierarchical_firewall_policy/main.tf (hunk 1 above)
Comment (renato-rudnicki) [reply 3-networks-hub-and-spoke/modules/hierarchical_firewall_policy/main.tf:2]: No reason, Probably a mistake. Fixing.

File: 3-networks-dual-svpc/modules/hierarchical_firewall_policy/versions.tf [hunk 2]
```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 3.50"
+      version = ">= 3.82"
```
Comment (apeabody) [inline 3-networks-dual-svpc/modules/hierarchical_firewall_policy/versions.tf:22]: Thanks - Can you also update for hub-and-spoke?

Comment (renato-rudnicki) [reply 3-networks-dual-svpc/modules/hierarchical_firewall_policy/versions.tf:22]: sure, my bad

**Review by apeabody — APPROVED** [review]
Thanks @renato-rudnicki! LGTM, just waiting on the CI

CC: @amandakarina

**Pull request merged**